stacks-common/src/util/secp256k1.rs

```diff
@@ -621,18 +630,16 @@ mod tests {
                 (Err(e1), Err(e2)) => assert_eq!(e1, e2),
                 (Err(e1), _) => {
                     test_debug!("Failed to verify signature: {}", e1);
-                    eprintln!(
```

Comment:
The reason this is `eprintln!()` is because we install a custom panic handler that, when run, can cause the message within the `panic!(..)` to get lost.  These changes will need to be reverted.

Reply:
Ah good to know. I will revert.

Reply:
Is that true in tests? I don't recall missing panic messages in test executions. I suspect that these were added when creating the fixtures -- the asserts were probably commented out so that a single execution would always test every fixture even if one failed.

Reply:
We have a custom panic hook to (1) force the node to print a backtrace (this doesn't always happen by default, it turns out), and (2) to force a core dump on unix-like systems by sending `SIGQUIT` to itself on panic.  However, for reasons I haven't figured out yet, the custom panic hook stops the panic message from being written.

Reply:
Is that custom hook active in unit tests? When I altered one of the fixtures here and run `cargo test util::secp256k1` in `stacks-common`, I see:

```
$ cargo test util::secp256k1
    Blocking waiting for file lock on build directory
   Compiling stacks-common v0.0.1 (/Users/aaron/devel/stacks-blockchain/stacks-common)
    Finished test [optimized + debuginfo] target(s) in 7.63s
warning: the following packages contain code that will be rejected by a future version of Rust: rstest v0.11.0
note: to see what the problems were, use the option `--future-incompat-report`, or run `cargo report future-incompatibilities --id 1`
     Running unittests src/libcommon.rs (/Users/aaron/devel/stacks-blockchain/target/debug/deps/stacks_common-50c7257d9fc9067d)

running 5 tests
test util::secp256k1::tests::test_verify_benchmark_roundtrip ... ignored
test util::secp256k1::tests::test_parse_serialize_compressed ... ok
test util::secp256k1::tests::test_parse_serialize ... ok
test util::secp256k1::tests::sk_from_seed ... ok
test util::secp256k1::tests::test_verify ... FAILED

failures:

---- util::secp256k1::tests::test_verify stdout ----
thread 'util::secp256k1::tests::test_verify' panicked at 'failed fixture (verification: Ok(false)): VerifyFixture {
    public_key: "0385f2e2867524289d6047d0d9c5e764c5d413729fc32291ad2c353fbc396a4219",
    data: "b84d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9",
    signature: "00354445a1dc98a1bd27984dbe69979a5cd77886b4d9134af5c40e634d96e1cb445b97de5b632582d31704f86706a780886e6e381bfed65228267358262d203fe6",
    result: Ok(
        true,
    ),
}', stacks-common/./src/util/secp256k1.rs:639:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    util::secp256k1::tests::test_verify

test result: FAILED. 3 passed; 1 failed; 1 ignored; 0 measured; 105 filtered out; finished in 0.00s

error: test failed, to rerun pass `--lib`
```

Reply:
ah so revert in integration tests? Leave in unit? Or revert in all cases? Cause there seems to be inconsitency throughout the code base as i already does panic!("MESSAGE") in a number of places but eprintln and unreachable or assert in others.

Reply:
It should revert in all cases.  Panics generally need to be accompanied by an `eprintln()`.

Reply:
Reverted `panic!(MESSAGE)` changes back to `assert!(false, MESSAGE)` and I silenced the warnings. I did not update any existing `panic!(MESSAGE)`'s though.